File: soundprocessing/src/main/cpp/Amp.cpp

#include "lefp"

class Amp {
private:
    int gain;

public:
    Amp(int gain) : gain(gain) {}

    Buffer process(Buffer *inputBuffer) {
        std::vector<float> samples = inputBuffer->getSamples();
        int N = samples.size();

        for (int i = 0; i < N; ++i) {
            for (int j = 0; j < 3; ++j) {
                samples[i] = lefp::larsenEquation(samples[i], gain);
            }
        }

        inputBuffer->setSamples(samples);

        return *inputBuffer;
    }

}

File: app/src/main/java/com/itis/guitarrecorder/utils/elements/Amp.kt
package com.itis.guitarrecorder.utils.elements

import com.itis.guitarrecorder.utils.ChainElement
import java.nio.Buffer


external fun processAmp(buffer: Buffer, gain: Int): Buffer
class Amp(
    val gain: Int
): ChainElement {
    override fun process(buffer: Buffer): Buffer {
        return processAmp(buffer, gain)
    }

}